### PR TITLE
[Bug] Honor vector_store upload limits in the files API

### PR DESCRIPTION
**Pull request by guan404ming**
Related #3329

## Purpose

- Make `/v1/files` honor `vector_store.max_file_size_mb` and `vector_store.supported_formats`; today both are accepted but the upload path hardcodes 50MB and a fixed extension list.
- Unset config keeps the previous defaults.

## Test Plan

- `go test ./pkg/apiserver/`

## Test Result

- Passed.

### CODE REVIEW

File: src/semantic-router/pkg/apiserver/route_files.go [hunk 1]
```diff
@@ -98,7 +123,7 @@ func (s *ClassificationAPIServer) handleUploadFile(w http.ResponseWriter, r *htt
 			return
 		}
 		s.writeErrorResponse(w, http.StatusBadRequest, "INVALID_INPUT",
-			fmt.Sprintf("failed to parse multipart form (max size: %dMB): %s", maxUploadSize/(1024*1024), err.Error()))
+			fmt.Sprintf("failed to parse multipart form (max size: %dMB): %s", maxBytes/(1024*1024), err.Error()))
```
Comment (AayushSaini101) [inline src/semantic-router/pkg/apiserver/route_files.go]: Thanks @guan404ming can we simply this like below, to avoid any overflow thanks : )

```
const MB int64 = 1024 * 1024
maxMB := maxBytes / MB
```

Comment (AayushSaini101) [reply src/semantic-router/pkg/apiserver/route_files.go]: and add test case to understand what will happen during the overflow process thanks

**Review by Xunzhuo — DISMISSED** [review]
Thanks for wiring these settings through. `uploadLimits()` reads `s.config` directly instead of the synchronized live-config path. Because `publishConfigMutation()` replaces that pointer under `configMu`, a concurrent file upload can race a management update; with a resolver-backed runtime config, the Files API can also keep enforcing the stale startup limits after reload. Please take one `cfg := s.currentConfig()` snapshot before reading `VectorStore` and add live-update/race coverage. The existing size-conversion overflow thread also remains open and needs to be resolved before approval.

**Comment by guan404ming** [comment]
Thanks! `uploadLimits()` now takes one `currentConfig()` snapshot, with resolver and concurrent-mutation tests. MB conversion falls back to the default on overflow, with a test.

File: src/semantic-router/pkg/apiserver/route_files.go (hunk 1 above)
Comment (guan404ming) [reply src/semantic-router/pkg/apiserver/route_files.go]: Thanks! Moved the conversion into `uploadMaxBytes()` with a `megabyte` constant and an overflow test.

**Review by AayushSaini101 — DISMISSED** [review]

File: src/semantic-router/pkg/apiserver/route_files.go (hunk 1 above)
Comment (AayushSaini101) [reply src/semantic-router/pkg/apiserver/route_files.go]: Thanks : ) @guan404ming

**Review by Xunzhuo — CHANGES_REQUESTED** [review]
The live-config and overflow fixes address the prior review, but the configured file-size boundary is still wrong: `MaxBytesReader` limits the entire multipart request, so a file exactly `max_file_size_mb` is rejected once multipart headers and boundaries are counted. Please enforce `maxBytes` against the file payload itself and add an exact-limit acceptance test.

**Review by FAUST-BENCHOU — COMMENTED** [review]
`supported_formats` can now make `.pdf` uploads succeed (and the reference config includes `.pdf`), but `vectorstore.ExtractText` still rejects `.pdf`. This means an accepted file will later fail every vector-store ingestion with `parse_error`.

Should PDF parsing be supported before exposing it here, or should this allowlist be constrained to formats the ingestion parser can actually consume?

**Review by wilsonwu — DISMISSED** [review]
LGTM, thanks!

**Review by Xunzhuo — DISMISSED** [review]
Three upload-contract blockers remain on this head. route_files.go applies the configured limit to the whole multipart envelope, so an exactly-at-limit file is rejected; the route wrapper still caps the body at 50 MiB, so the shipped 100 MiB setting cannot work; and the accepted .pdf format reaches vectorstore.ExtractText, which rejects PDF. Please enforce the configured limit on file bytes with a dynamic envelope guard, add exact-limit and over-50-MiB regressions, and align accepted formats with parser support or implement PDF ingestion.

**Review by Xunzhuo — APPROVED** [review]
